commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/PaperCommandRegistration.java

```diff
@@ -89,9 +157,77 @@ public void unregister(String commandName, boolean unregisterNamespaces, boolean
 
 	@Override
 	public void preReloadDataPacks() {
-		RootCommandNode<Source> root = getBrigadierDispatcher.get().getRoot();
+		RootCommandNode<Source> root = getPaperDispatcher().getRoot();
 		for (CommandNode<Source> commandNode : registeredNodes.getChildren()) {
 			root.addChild(commandNode);
 		}
+		reloadHelpTopics.run();
+		CommandAPIBukkit.get().updateHelpForCommands(CommandAPI.getRegisteredCommands());
 	}
+
+	@SuppressWarnings("unchecked")
+	private LiteralCommandNode<Source> asPluginCommand(LiteralCommandNode<Source> commandNode) {
+		try {
+			if (pluginCommandNodeConstructor.getDeclaringClass().getSimpleName().equals("PluginCommandNode")) {
+				return (LiteralCommandNode<Source>) pluginCommandNodeConstructor.newInstance(
+					commandNode.getLiteral(),
+					CommandAPIBukkit.getConfiguration().getPlugin().getPluginMeta(),
+					commandNode,
+					getDescription(commandNode.getLiteral())
+				);
+			} else {
+				setPluginCommandMeta(commandNode);
+				return commandNode;
+			}
+		} catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private void setPluginCommandMeta(LiteralCommandNode<Source> node) {
+		try {
+			Field metaField = node.getClass().getSuperclass().getDeclaredField("pluginCommandMeta");
+			metaField.setAccessible(true);
+			metaField.set(node, pluginCommandNodeConstructor.newInstance(
+				CommandAPIBukkit.getConfiguration().getPlugin().getPluginMeta(),
+				getDescription(node.getLiteral()),
+				getAliasesForCommand(node.getLiteral())
+			));
+		} catch (NoSuchFieldException | InvocationTargetException | InstantiationException | IllegalAccessException e) {
+			// This doesn't happen
+		}
+	}
+
+	private String getDescription(String commandName) {
+		String description = "";
+		for (RegisteredCommand command : CommandAPI.getRegisteredCommands()) {
+			String namespaceStripped = "";
+			if (commandName.contains(":")) {
+				namespaceStripped = commandName.split(":")[1];
+			} else {
+				namespaceStripped = commandName;
+			}
+			if (command.commandName().equals(namespaceStripped) || Arrays.asList(command.aliases()).contains(namespaceStripped)) {
+				Object helpTopic = command.helpTopic().orElse(null);
+				if (helpTopic != null) {
+					description = ((HelpTopic) helpTopic).getShortText();
```

Comment:
Not sure if this is the best way to use an `Optional`. Should probably use the fact that `command.helpTopic()` is an `Optional` rather than converting it to `null`.
```suggestion
				Optional<Object> helpTopic = command.helpTopic();
				if (helpTopic.isPresent()) {
					description = ((HelpTopic) helpTopic.get()).getShortText();
				} else {
```

Reply:
Not sure if this is worth it though. I mean, we don't reduce code or make this more efficient so yeah, not so sure about this.